OpenRA.Game/Graphics/HardwarePalette.cs

```diff
@@ -16,14 +16,15 @@ namespace OpenRA.Graphics
 {
 	public sealed class HardwarePalette : IDisposable
 	{
-		public const int MaxPalettes = 256;
+		const int MaxPalettes = 256;
 
 		public ITexture Texture { get; private set; }
+		public int Height { get { return buffer.GetLength(0); } }
 		readonly Dictionary<string, ImmutablePalette> palettes = new Dictionary<string, ImmutablePalette>();
 		readonly Dictionary<string, MutablePalette> modifiablePalettes = new Dictionary<string, MutablePalette>();
 		readonly IReadOnlyDictionary<string, MutablePalette> readOnlyModifiablePalettes;
 		readonly Dictionary<string, int> indices = new Dictionary<string, int>();
-		readonly uint[,] buffer = new uint[MaxPalettes, Palette.Size];
+		uint[,] buffer = new uint[0, Palette.Size];
```

Comment:
Would it make sense to define a sensibly-sized default (say 8 or 16) to cut down on a few resizes?  Probably not in practice...